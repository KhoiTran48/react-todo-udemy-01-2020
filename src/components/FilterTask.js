import React, { Component } from 'react'
import {setFilterTextAction, setOrderAction, setFiltertDateAction} from '../actions/filterAction'
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import moment from 'moment';

class FilterTask extends Component {
    constructor(props){
        super(props);
        let startDate = props.filter.startDate ? moment(props.filter.startDate) : moment()
        let endDate = props.filter.endDate ? moment(props.filter.endDate) : moment()
        let text = props.filter.text
        let order = props.filter.order
        this.state = {startDate, endDate, text, order, focusedInput: null}
    }
    handleDate = (startDate, endDate)=>{
        if(startDate && endDate){
            this.setState({startDate, endDate})
            this.props.setFiltertDate(startDate.valueOf(), endDate.valueOf())
        }
    }
    handleInput = (e) => {
        this.setState({text: e.target.value})
        this.props.setFilterText(e.target.value)
    }
    handleSelect = (e) => {
        this.setState({order: e.target.value})
        this.props.setOrder(e.target.value)
    }
    render() {
        return (
            <div>
                <input type="text" name="filterTitle" 
                    onChange={this.handleInput} 
                    value={this.state.text}/>
                <select onChange={this.handleSelect} value={this.state.order}>
                    <option value="asc">Deadline ASC</option>
                    <option value="desc">Deadline DESC</option>
                </select>
                <DateRangePicker
                    startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                    startDateId="your_unique_start_date_id"
                    endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                    endDateId="your_unique_end_date_id"
                    onDatesChange={({ startDate, endDate }) => this.handleDate(startDate, endDate)} // PropTypes.func.isRequired,
                    focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
                    numberOfMonths={1}
                />
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        filter: state.filter
    }
}

const mapDispatchToProps = dispatch => {
    return {
        setFilterText: (text) => dispatch(setFilterTextAction(text)),
        setOrder: (order) => dispatch(setOrderAction(order)),
        setFiltertDate: (startDate, endDate) => dispatch(setFiltertDateAction(startDate, endDate))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(FilterTask)
