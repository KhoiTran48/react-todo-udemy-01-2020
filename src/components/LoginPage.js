import React, { Component } from 'react'
import {connect} from 'react-redux'
import {signIn} from '../actions/auth'

class LoginPage extends Component {
    render() {
        return (
            <div>
                <button onClick={this.props.signIn}>Login</button>
            </div>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        signIn: ()=> dispatch(signIn())
    }
}

export default connect(null, mapDispatchToProps)(LoginPage)
