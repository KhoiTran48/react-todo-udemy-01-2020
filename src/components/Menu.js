import {NavLink} from 'react-router-dom'
import React, { Component } from 'react'

export default class Menu extends Component {
    render() {
        return (
            <div>
                <NavLink to="/" exact>Dashboard</NavLink>
                <NavLink to="/add">Add Task</NavLink>
            </div>
        )
    }
}
