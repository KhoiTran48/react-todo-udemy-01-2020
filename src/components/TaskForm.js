import React, { Component } from 'react'
import moment from 'moment'
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';


export default class TaskForm extends Component {
    constructor(props){
        super(props)
        this.state={
            taskId: props.taskId ? props.taskId : "",
            title : props.title ? props.title : "",
            desc : props.desc ? props.desc : "",
            deadline: props.deadline ? moment(props.deadline) : moment() 
        }
    }

    handleInput = (e)=>{
        if(e.target.value){
            this.setState({[e.target.name]: e.target.value})
        }
    }

    handleSubmit = (e) =>{
        let deadline = this.state.deadline.valueOf();
        this.props.handleSubmit({...this.state, deadline})
    }

    handleDate = (deadline) => {
        if(deadline){
            this.setState({deadline})
        }
    }

    render() {
        return (
            <div>
                Title
                <input 
                    type="text" 
                    name="title" 
                    value={this.state.title}
                    onChange={this.handleInput}
                    autoFocus
                    />
                Desc
                <input 
                    type="text" 
                    name="desc" 
                    value={this.state.desc}
                    onChange={this.handleInput}
                    />
                Created At
                <SingleDatePicker
                    date={this.state.deadline} // momentPropTypes.momentObj or null
                    onDateChange={this.handleDate} // PropTypes.func.isRequired
                    focused={this.state.focused} // PropTypes.bool
                    onFocusChange={({ focused }) => this.setState({ focused })} // PropTypes.func.isRequired
                    numberOfMonths={1}
                    />
                <button onClick={this.handleSubmit}>Save</button>
            </div>
        )
    }
}
