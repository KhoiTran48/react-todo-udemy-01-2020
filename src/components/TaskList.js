import React, { Component } from 'react'
import {connect} from 'react-redux'
import TaskInfo from './TaskInfo'
import selectTask from '../selectors/tasks'

export class TaskList extends Component {
    constructor(props){
        super(props)
    }
    renderTask(listTask){
        return listTask.map((task, key)=>{
            return <TaskInfo key={key} {...task}/>
        })
    }
    render() {
        return (
            <div>
                {this.renderTask(this.props.listTask)}
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        listTask: selectTask(state.listTask, state.filter)
    }
}

export default connect(mapStateToProps)(TaskList)