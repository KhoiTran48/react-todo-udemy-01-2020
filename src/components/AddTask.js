import React, { Component } from 'react'
import TaskForm from './TaskForm'
import {addTaskAction} from '../actions/taskAction'
import {connect} from 'react-redux'

class AddTask extends Component {
    render() {
        return (
            <div>
                <TaskForm
                    handleSubmit = {(taskForm)=>{
                        this.props.addTaskAction(taskForm)
                        this.props.history.push("/")
                    }}                
                />
            </div>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        addTaskAction: (task) => dispatch(addTaskAction(task))
    }
}

export default connect(null, mapDispatchToProps)(AddTask)