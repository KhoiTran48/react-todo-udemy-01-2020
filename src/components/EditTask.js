import React, { Component } from 'react'
import TaskForm from './TaskForm'
import {connect} from 'react-redux'
import {editTaskAction, deleteTaskAction} from '../actions/taskAction'


class EditTask extends Component {
    render() {
        if(!this.props.editTask){
            this.props.history.push("/")
        }
        return (
            <div>
                <TaskForm
                    handleSubmit = {(taskForm)=>{
                        this.props.editTaskAction(taskForm)
                        this.props.history.push("/")
                    }}  
                    {...this.props.editTask}
                />
                <button
                    onClick={()=>{
                        this.props.deleteTaskAction(this.props.editTask.taskId)
                        this.props.history.push("/")
                    }}
                >Delete</button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        editTask : state.listTask.find(task=>task.taskId === props.match.params.id)
    }
}

const mapDispatchToProps = dispatch => {
    return {
        editTaskAction: (task) => dispatch(editTaskAction(task)),
        deleteTaskAction: (taskId) => dispatch(deleteTaskAction(taskId))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditTask)
