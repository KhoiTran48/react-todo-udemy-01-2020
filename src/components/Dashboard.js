import React, { Component } from 'react'
import TaskList from './TaskList'
import FilterTask from './FilterTask'

class Dashboard extends Component {
    render() {
        return (
            <div>
                <FilterTask></FilterTask>
                <TaskList></TaskList>
            </div>
        )
    }
}

export default Dashboard



