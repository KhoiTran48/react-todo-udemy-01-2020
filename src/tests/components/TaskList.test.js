import React from 'react'
import {TaskList} from '../../components/TaskList'
import {shallow} from 'enzyme'
import listTask from '../fixtures/listTask'

test("render task list", ()=>{
    const wrapper = shallow(<TaskList listTask={listTask}/>)
    expect(wrapper).toMatchSnapshot();
})