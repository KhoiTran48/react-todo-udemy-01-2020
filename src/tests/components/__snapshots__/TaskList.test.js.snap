// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render task list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskList
    listTask={
        Array [
            Object {
              "deadline": 0,
              "desc": "",
              "taskId": 1,
              "title": "JS",
            },
            Object {
              "deadline": 1000,
              "desc": "",
              "taskId": 2,
              "title": "PHP",
            },
            Object {
              "deadline": 2000,
              "desc": "",
              "taskId": 3,
              "title": "ASP",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TaskInfo
          deadline={0}
          desc=""
          taskId={1}
          title="JS"
/>,
        <TaskInfo
          deadline={1000}
          desc=""
          taskId={2}
          title="PHP"
/>,
        <TaskInfo
          deadline={2000}
          desc=""
          taskId={3}
          title="ASP"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "deadline": 0,
          "desc": "",
          "taskId": 1,
          "title": "JS",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "deadline": 1000,
          "desc": "",
          "taskId": 2,
          "title": "PHP",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "deadline": 2000,
          "desc": "",
          "taskId": 3,
          "title": "ASP",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TaskInfo
            deadline={0}
            desc=""
            taskId={1}
            title="JS"
/>,
          <TaskInfo
            deadline={1000}
            desc=""
            taskId={2}
            title="PHP"
/>,
          <TaskInfo
            deadline={2000}
            desc=""
            taskId={3}
            title="ASP"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "deadline": 0,
            "desc": "",
            "taskId": 1,
            "title": "JS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "deadline": 1000,
            "desc": "",
            "taskId": 2,
            "title": "PHP",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "deadline": 2000,
            "desc": "",
            "taskId": 3,
            "title": "ASP",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
